pkg/resources/password_policy.go

```diff
@@ -206,32 +206,51 @@ func ReadPasswordPolicy(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return err
 	}
-	if err := d.Set("min_length", passwordPolicyDetails.PasswordMinLength.Value); err != nil {
-		return err
+	if passwordPolicyDetails.PasswordMinLength != nil && passwordPolicyDetails.PasswordMinLength.Value != nil {
```

Comment:
Ok, I did write such  functions, I think the code looks cleaner now.

Reply:
this is a good idea. Another popular strategy is to create a struct with flatten / expand functions. The "flatten" function saves it to TF state, while the expand function instantiates the struct from TF state. It could be combined with the function here I think.

Reply:
Okay, my part of the comment is fine, but I am not resolving it because of the above comment which awaits @sfc-gh-swinkler answer.